### PR TITLE
Fix misc TypeScript errors in client 9

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This fixes minor TS errors in `client`.

#### Testing instructions

This should not change any behavior.

### CODE REVIEW

File: client/my-sites/marketplace/components/index.tsx [hunk 1]
```diff
@@ -1,16 +1,8 @@
 import { Button } from '@automattic/components';
 import styled from '@emotion/styled';
-import { HorizontalRule } from '@wordpress/components';
-import { MarketplaceThemeProps } from 'calypso/my-sites/marketplace/theme';
-
-export const MobileHiddenHorizontalRule = styled( HorizontalRule )< MarketplaceThemeProps >`
```
Comment (sirbrillig) [inline client/my-sites/marketplace/components/index.tsx:6]: This is unused code (and the `tabletDown` prop had been removed already).

File: client/my-sites/marketplace/components/index.tsx [hunk 2]
```diff
@@ -1,16 +1,8 @@
...
-	@media ( ${ ( { theme } ) => theme?.breakpoints.tabletDown } ) {
-		display: none;
-	}
-`;
 
 export const MarketplaceHeaderTitle = styled.h1`
-	font-size: ${ ( { subtitle = false } ) => ( subtitle ? '1.5em' : '2em' ) };
```
Comment (sirbrillig) [inline client/my-sites/marketplace/components/index.tsx:13]: The `subtitle` prop is not used by anything that renders this.

File: client/my-sites/marketplace/pages/marketplace-plugin-install/index.tsx [hunk 3]
```diff
@@ -60,9 +66,14 @@ const MarketplacePluginInstall = ( {
 		getUploadedPluginId( state, siteId )
 	) as string;
 	const pluginUploadComplete = useSelector( ( state ) => isPluginUploadComplete( state, siteId ) );
-	const installedPlugin = useSelector( ( state ) =>
```
Comment (sirbrillig) [inline client/my-sites/marketplace/pages/marketplace-plugin-install/index.tsx:63]: For some reason this cannot find the correct type from the selector, so we hard-code it here by starting from `unknown` and using type guards. At some point this should be done properly by fixing the selector's return type.

**Comment by sirbrillig** [comment]
Fixes https://github.com/Automattic/wp-calypso/issues/59605

File: client/my-sites/marketplace/pages/marketplace-plugin-install/index.tsx (hunk 3 above)
Comment (sirbrillig) [reply client/my-sites/marketplace/pages/marketplace-plugin-install/index.tsx:63]: Turns out this did not work if the selector returned undefined. https://github.com/Automattic/wp-calypso/pull/59608 should fix it in a safer way.

File: client/components/signup-site-preview/screenshot.tsx [hunk 4]
```diff
@@ -52,18 +52,22 @@ export default function SignupSitePreviewScreenshot( {
...
+						? String(
+								translate( 'Preview of site with phone layout', {
+									comment: 'alt text of site preview',
+								} )
+						  )
+						: String(
+								translate( 'Preview of site with desktop layout', {
+									comment: 'alt text of site preview',
+								} )
+						  )
```
Comment (daledupreez) [inline client/components/signup-site-preview/screenshot.tsx:70]: Minor note: could you also supply `textOnly: true` in options to trigger a string return type? This is something that was added recently in #58938.

**Review by daledupreez — COMMENTED** [review]
Thanks for tackling this cleanup, @sirbrillig! 🙇 💖

File: client/components/signup-site-preview/screenshot.tsx (hunk 4 above)
Comment (sirbrillig) [reply client/components/signup-site-preview/screenshot.tsx:70]: Yes, totally. I just learned about that myself.

**Pull request merged**